libntxm/common/source/sample.cpp

```diff
@@ -241,22 +238,39 @@ void Sample::play(u8 note, u8 volume_ , u8 channel)
 	SCHANNEL_CR(channel) = 0;
 	SCHANNEL_TIMER(channel) = SOUND_FREQ((int)LOOKUP_FREQ(realnote,finetune));
 
-	if( loop == NO_LOOP )
+	u32 offs_samps = FT_OFFSET_SCALAR * offs * (sound_format == SOUNDXCNT_FORMAT_8BIT ? 1 : 2);
```

Comment:
Round the sample offset up to a multiple of four. Otherwise, when it's not one, SCHANNEL_LENGTH will skip some samples at the end relative to SCHANNEL_SOURCE.

Reply:
The offset in samples is always a multiple of four, as it's calculated with a multiplication by 256